kong/plugins/opentelemetry/proto.lua

```diff
@@ -0,0 +1,481 @@
+-- Copyright 2019, OpenTelemetry Authors
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+--- Proto
+-- defines the OTLP .proto
+local protoc = require "protoc".new()
+
+local function load_proto(filename, data)
+  assert(protoc:load(data, filename))
+end
+
+-- parse & load proto files globally on module loading
+-- https://github.com/open-telemetry/opentelemetry-proto/blob/v0.18.0/opentelemetry/proto/common/v1/common.proto
+load_proto("opentelemetry/proto/common/v1/common.proto", [[
```

Comment:
consider move this and following to `kong/include/opentelemetry/proto/common/v1/common.proto`

Reply:
awesome